dev-ostelco-ios-clientTests/CoordinatorTests/EdgeCasesInStageDeciderTests.swift

```diff
@@ -54,5 +54,97 @@ class EdgeCasesInStageDeciderTests: XCTestCase {
         
         XCTAssertEqual(decider.compute(context: context, localContext: localContext), .home)
     }
+    
+    func testServerIsUnavailableOnStartUp() {
+        let decider = StageDecider()
+        let localContext = LocalContext(serverIsUnreachable: true)
+        
+        XCTAssertEqual(decider.compute(context: nil, localContext: localContext), .ohNo(.noInternet))
+    }
+    
+    // Edge cases for Singapore flow
+    
+    func testUserHasSelectedACountryAndHasLocationProblem() {
+        let decider = StageDecider()
+        let localContext = LocalContext(selectedRegion: Region(id: "sg", name: "SG"), hasLocationProblem: true)
+        let context = Context(customer: CustomerModel(id: "xxx", name: "xxx", email: "xxxx@gmail.com", analyticsId: "xxxx", referralId: "xxxx"), regions: [])
+        
+        XCTAssertEqual(decider.compute(context: context, localContext: localContext), .locationProblem)
+    }
+    
+    func testUserHasSelectedSingpassAndCancelledSingpass() {
+        let decider = StageDecider()
+        let localContext = LocalContext(selectedRegion: Region(id: "sg", name: "SG"), regionVerified: true, hasSeenVerifyIdentifyOnboarding: true)
+        
+        let context = Context(customer: CustomerModel(id: "xxx", name: "xxx", email: "xxxx@gmail.com", analyticsId: "xxxx", referralId: "xxxx"), regions: [])
+        
+        XCTAssertEqual(decider.compute(context: context, localContext: localContext), .selectIdentityVerificationMethod([.scanIC, .singpass]))
+    }
+    
+    func testUserHasCompletedNRICAndCancelledJumioInSingapore() {
+        let decider = StageDecider()
+        let localContext = LocalContext(selectedRegion: Region(id: "sg", name: "SG"), regionVerified: true, hasSeenVerifyIdentifyOnboarding: true)
+        
+        let context = Context(
+            customer: CustomerModel(id: "xxx", name: "xxx", email: "xxxx@gmail.com", analyticsId: "xxxx", referralId: "xxxx"),
+            regions: [
+                RegionResponse(
+                    region: Region(id: "sg", name: "Singapore"),
+                    status: .PENDING,
```

Comment:
If we wait long enough, jumio will send a rejection callback to the server with reason "no id uploaded". This will mark jumio as `rejected`.

Reply:
This would be covered by our tests to handle being rejected by Jumio. This test here is for what happens when the user kills the app in the middle of the Jumio and starts the app again. Before Jumio has rejected the user.